neo/Ledger/MemoryPool.cs

```diff
@@ -357,10 +374,9 @@ internal void UpdatePoolForBlockPersisted(Block block, Snapshot snapshot)
 
             // If we know about headers of future blocks, no point in verifying transactions from the unverified tx pool
             // until we get caught up.
-            if (block.Index > 0 && block.Index < Blockchain.Singleton.HeaderHeight)
+            if (block.Index > 0 && block.Index < Blockchain.Singleton.HeaderHeight || _policyChanged)
```

Comment:
Better to add some parenthesis here... not absolutely clear where the OR applies.

Reply:
AND (&&) has a higher priority than OR (||).